### PR TITLE
Add force kill to docker job

### PR DESCRIPTION
**Pull request by YanzhaoLi**
**What this PR does / why we need it**:

**How can this PR be verified?**
https://pks.ci.cf-app.com/teams/bosh-lifecycle-dev/pipelines/pks-api-1.10.x-add-force-kill-to-docker-job
**Is there any change in kubo-deployment?**
No
**Is there any change in kubo-ci?**
No
**Does this affect upgrade, or is there any migration required?**
No
**Which issue(s) this PR fixes:**
https://pivotal-spike.atlassian.net/browse/PKS-858

**Release note**:

```release-note

```

### CODE REVIEW

File: jobs/docker/templates/bin/ctl [hunk 1]
```diff
@@ -124,6 +142,15 @@ case $1 in
       set +e
       echo "Unmounting /var/vcap/store/docker/docker"
       umount /var/vcap/store/docker/docker
+      # Check umount status
+      if [ ! -z $(cat /proc/mounts | grep /var/vcap/store/docker/docker) ]; then
```
Comment (lubronzhan) [inline jobs/docker/templates/bin/ctl:146]:
```
exit_status=$?
if [ $exit_status -eq 0 ]; then
```

File: jobs/docker/templates/bin/ctl [hunk 2]
```diff
@@ -99,6 +99,24 @@ case $1 in
...
+    containers="$(/var/vcap/packages/docker/bin/docker --host unix://${DOCKER_PID_DIR}/docker.sock ps -q | head -n 50)"
+    if [ ! -z "$containers" ]; then
+    # Let's wait grace-period time for docker stop finish.
+      echo "Kill the leftover containers: $containers"
+      timeout 60 killall -I containerd-shim --wait ||  "Fail to kill leftover containers in 60s"
+    fi
+
+    pids=$(pgrep containerd-shim)
+    if [ ! -z "$pids" ]; then
+      echo "Need force killing leftover containers: $(ps aux | grep [c]ontainerd-shim)"
```
Comment (lubronzhan) [inline jobs/docker/templates/bin/ctl:113]: typo: [c]ontainerd-shim)

**Comment by yuyangbj** [comment]
Ship it, no comment from me.

File: jobs/docker/templates/bin/ctl [hunk 3]
```diff
@@ -124,6 +142,15 @@ case $1 in
       set +e
       echo "Unmounting /var/vcap/store/docker/docker"
       umount /var/vcap/store/docker/docker
+      # Check umount status
+      if [ ! -z $(cat /proc/mounts | grep /var/vcap/store/docker/docker) ]; then
+        echo "Fail to umount /var/vcap/store/docker/docker. Processes using this fs: "
+        fuser -M /var/vcap/store/docker/docker
```
Comment (lubronzhan) [inline jobs/docker/templates/bin/ctl:148]: set +x so we could see what command is being used. Or print fuser and lsof

Comment (lubronzhan) [reply jobs/docker/templates/bin/ctl:148]: Print /proc/mount

File: jobs/docker/templates/bin/ctl (hunk 1 above)
Comment (YanzhaoLi) [reply jobs/docker/templates/bin/ctl:146]: If not mounted, the return code ` umount /var/vcap/store/docker/docker` would also not be zero

File: jobs/docker/templates/bin/ctl (hunk 2 above)
Comment (YanzhaoLi) [reply jobs/docker/templates/bin/ctl:113]: To avoid displaying the process of `grep` itself

File: jobs/docker/templates/bin/ctl (hunk 3 above)
Comment (YanzhaoLi) [reply jobs/docker/templates/bin/ctl:148]: We've already in the `set +x`

File: jobs/docker/templates/bin/ctl (hunk 1 above)
Comment (lubronzhan) [reply jobs/docker/templates/bin/ctl:146]: ok

File: jobs/docker/templates/bin/ctl (hunk 2 above)
Comment (lubronzhan) [reply jobs/docker/templates/bin/ctl:113]: ok

File: jobs/docker/templates/bin/ctl (hunk 3 above)
Comment (lubronzhan) [reply jobs/docker/templates/bin/ctl:149]: ok

**Pull request merged**